server/src/main/java/org/opensearch/cluster/RepositoryCleanupInProgress.java

```diff
@@ -101,13 +104,46 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.startObject();
             {
                 builder.field("repository", entry.repository);
+                if (params.param(Metadata.CONTEXT_MODE_PARAM, Metadata.CONTEXT_MODE_API).equals(Metadata.CONTEXT_MODE_GATEWAY)) {
```

Comment:
What about the BWC for this?

Reply:
By default when we are using toXContent API mode is passed, we only pass Gateway when we are saving cluster state as persisted state in remote. So, by default I am putting the params as API and only adding extra fields if the mode is Gateway mode which is only used in our use case.